spring-kafka-docs/src/main/antora/modules/ROOT/pages/testing.adoc

```diff
@@ -404,6 +404,15 @@ public class MyApplicationTests {
 
 Notice that, since this is a Spring Boot application, we override the broker list property to set Spring Boot's property.
 
+[[embedded-broker-with-springjunitconfig-annotations]]
+== @EmbeddedKafka with `@SpringJunitConfig`
+
+When using `@EmbeddedKafka` with `@SpringJUnitConfig`, it is recommended to use `@DirtiesContext` on the test class.
+This is to prevent potential race conditions occurring during the JVM shutdown after running multiple tests in a test suite.
+For example, without using `@DirtiesContext`, the `EmbeddedKafkaBroker` may shutdown earlier while the application context still needs resources from it.
+Adding `@DirtiesContext` will ensure that the application context is cleaned up after each test and not cached, making it less vulnerable to potential resource race conditions like these.
```

Comment:
We probably still need to say that error in logs is harmful since every new embedded broker run creates its own temporary directory.